src/main/java/io/wispforest/owo/ui/component/TextAreaComponent.java

```diff
@@ -154,6 +158,23 @@ public TextAreaComponent text(String text) {
         return this;
     }
 
+    public TextTransform transform() {
+        return transform;
+    }
+
+    public TextAreaComponent transform(TextTransform transform) {
+        this.transform = transform;
+        checkTransform(getText());
+        return this;
+    }
+
+    protected void checkTransform(String value) {
+        String transformed = transform.apply(value);
+        if (!transformed.equals(value)) {
+            setText(transformed);
```

Comment:
Won't doing it like this effectively mean that the `onChanged` event get sent twice?